hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -109,6 +123,29 @@ private void validateIdentifier(String id, Set<String> identifiers, String confi
     }
   }
 
+  private Option<Schema> validateAndGetTransformedSchema(TransformerInfo transformerInfo, Dataset<Row> dataset, Option<Schema> incomingSchemaOpt,
+                                                         JavaSparkContext jsc, SparkSession sparkSession, TypedProperties properties) {
+    if (!incomingSchemaOpt.isPresent()) {
+      throw new HoodieSchemaException(String.format("Source schema not available for transformer %s", transformerInfo));
+    }
+    Schema incomingSchema = incomingSchemaOpt.get();
+
+    // Get expected target schema from transformer and actual target schema from struct
+    Schema targetSchema = AvroConversionUtils.convertStructTypeToAvroSchema(dataset.schema(), incomingSchema.getName(),
+        incomingSchema.getNamespace());
+    Option<Schema> expectedTargetSchemaOpt = transformerInfo.getTransformer().transformedSchema(jsc, sparkSession, incomingSchema, properties);
+    if (!expectedTargetSchemaOpt.isPresent()) {
+      throw new HoodieSchemaException(String.format("Expected target schema not provided for transformer %s", transformerInfo));
+    }
+
+    Schema expectedTargetSchema = expectedTargetSchemaOpt.get();
+    if (!AvroSchemaUtils.isSchemaCompatible(expectedTargetSchema, targetSchema, false, false)) {
```

Comment:
are we looking for schema equality or compatability?
if its equality, then probably you should do 
```
!AvroSchemaUtils.isSchemaCompatible(expectedTargetSchema, targetSchema, false, false) || !AvroSchemaUtils.isSchemaCompatible(targetSchema, expectedTargetSchema, false, false)
```

Reply:
I tried this but it is leading to test failures. This is because for one of the fields - schema converted from struct type is `["null","long"]` and expected schema is `"long"`. Therefore expected schema can not be converted to the schema converted from struct.

Reply:
ok. 
can we make the checkNaming to true though? (3rd arg)

Reply:
I can make that change but then it will match the schema namespace as well (This could be error prone if transformer implementation doesn't make sure namespace is retained). Currently the tests are failing after enabling it. Will debug those.